front/pages/api/stripe/webhook.ts

```diff
@@ -189,7 +190,7 @@ async function handler(
               // We block a new subscription if the active one is with payment
               if (
                 activeSubscription &&
-                activeSubscription.plan.stripeProductId !== null
+                activeSubscription.plan.code === PRO_PLAN_SEAT_29_CODE
```

Comment:
here again assuming that only PRO PLAN is paid via credit card.

Reply:
It is debatable, maybe we'd want to block this flow if the active sub has a stripe subscription ID (so we'd also block enterprise plan workspaces from going through)

Reply:
Why not `if (activeSubscription && activeSubscription.stripeSubscriptionId !== null)`?

It should work since in case of upgrade to a paid plan from another paid plan we should have first set is as "ended_backend_only".